projects/packages/my-jetpack/src/class-initializer.php

```diff
@@ -606,4 +611,54 @@ public static function has_file_system_write_access() {
 	public static function get_idc_container_id() {
 		return static::IDC_CONTAINER_ID;
 	}
+
+	/**
+	 * Conditionally append the red bubble notification to the "Jetpack" menu item if there are alerts to show
+	 *
+	 * @return void
+	 */
+	public static function maybe_show_red_bubble() {
+		global $menu;
+		// filters for the items in this file
+		add_filter( 'my_jetpack_red_bubble_notification_slugs', array( __CLASS__, 'alert_if_missing_site_connection' ) );
+		$red_bubble_alerts = self::get_red_bubble_alerts();
+
+		// The Jetpack menu item should be on index 3
+		if ( ! empty( $red_bubble_alerts ) && isset( $menu[3] ) && $menu[3][0] === 'Jetpack' ) {
+			// phpcs:ignore WordPress.WP.GlobalVariablesOverride.Prohibited
+			$menu[3][0] .= sprintf( ' <span class="awaiting-mod">%d</span>', count( $red_bubble_alerts ) );
+		}
+	}
+
+	/**
+	 * Collect all possible alerts that we might use a red bubble notification for
+	 *
+	 * @return array|mixed|null
+	 */
+	public static function get_red_bubble_alerts() {
+		static $red_bubble_alerts = array();
+
+		// using a static cache since we call this function more than once in the class
+		if ( ! empty( $red_bubble_alerts ) ) {
+			return $red_bubble_alerts;
+		}
+		// go find the alerts
+		$red_bubble_alerts = apply_filters( 'my_jetpack_red_bubble_notification_slugs', $red_bubble_alerts );
+
+		return $red_bubble_alerts;
+	}
+
+	/**
+	 * Add an alert slug if the site is missing a site connection
+	 *
+	 * @param array $red_bubble_slugs - slugs that describe the reasons the red bubble is showing.
+	 * @return array
+	 */
+	public static function alert_if_missing_site_connection( $red_bubble_slugs ) {
```

Comment:
Since this function expects an array, does it make sense to declare the type here? Otherwise, it'll throw an error in the code below (line 659)
```suggestion
	public static function alert_if_missing_site_connection( array $red_bubble_slugs ) {
```